include/alpaka/core/Assert.hpp

```diff
@@ -20,8 +20,12 @@
 
 #ifdef ALPAKA_DEBUG_OFFLOAD_ASSUME_HOST
 #    define ALPAKA_ASSERT_OFFLOAD(EXPRESSION) ALPAKA_ASSERT(EXPRESSION)
+#elif defined __AMDGCN__ && (!defined NDEBUG)
+#    define ALPAKA_ASSERT_OFFLOAD(EXPRESSION)                                                                         \
+        if(!(EXPRESSION))                                                                                             \
+        __builtin_trap()
```

Comment:
Somehow this passes the clang-tidy check, but the indentation seems odd here... .

Maybe it is the missing semicolon at the end. If this is added empty expression warnings are raised because invocation sites add another one.

Should this stay this way?

Reply:
Would it work to put everything in one line here? Otherwise you can turn clang-format off for this particular line and indent by hand.

```suggestion
        // clang-format off
        if(!(EXPRESSION))                                                                                             \
            __builtin_trap()
        //clang-format on
```

Reply:
Having everything on one line was my first version, I think, but clang-format changed it.

The clang format comments would need to surround the whole macro definition, the suggestion breaks it. Maybe inline comments (`/* clang-format off */`) would work?

I am against turning off clang format in this place, because the style demands alignment of these line continuation way to the right, which is just too tedious without clang format.

Workable options would be:
1. Check if the clang-format style can be adjusted to fix this.
2. Adding  `-Wno-extra-semi-stmt` in cmake and adding a semicolon.
3. ignore it and move on

Reply:
Disabling clang format in edge cases is fine.

1. Changing the format only for an edge case is not the best way, you would always find some strange cases
2. Is not possible because we support stand allows header too and can not enforce the user to disable warnings

The common way to write a macro and enforce a semicolon is

```
#define MYMACRO do { \
  if(!(EXPRESSION))  \
      __builtin_trap(); \
while(false)
```

Reply:
That `while(false)` does not look save: Would this not swallow the next statement if there is no semicolon without raising an error? I guess some stand-alone no-op statement would be more safe...

Reply:
The inline disabling of clang-format should work, too. Maybe that is the best approach here.

Reply:
Ok, I put the inline clang-format switch in.

Reply:
`do {}while(false)` is the standard expression you use in such a case, it is a common pattern.
The advantage of the do-while is that is works well with if conditions too.

Reply:
That would work, does not swallow statements and raises an error if the semicolon is missing.

Reply:
I added the `do{} while(false)` version.